### PR TITLE
Update BYOA docs with diagrams and clean up organization

### PR DESCRIPTION
**Pull request by ak-aiven**

### CODE REVIEW

**Review by lornajane — COMMENTED** [review]
This definitely doesn't count as a review, just a few tips from my first glance! We have some lint checking built in so have a look at the comments from the bot as well as me. We're using mermaid for diagrams (yours are lovely but we won't be able to maintain or update them), do you think you could convert? Also if you do include images, these need alt tags (but the syntax check doesn't pick that up, it's on our list)

**Comment by ak-aiven** [comment]
Thanks @lornajane. I changed the casing for the sub headings so the linter is a little happier now. I have already invested quite a bit of time in those diagrams and I am not sure I have the bandwidth to convert at this time. Those diagrams are taken from Lucidchart which is where the security team maintains these architectural references.

**Comment by ak-aiven** [comment]
@encima do you want to convert these? :)

**Comment by lornajane** [comment]
I recommend we review this as-is, replacing the diagrams at a later date if at all possible.

**Review by dbeech — CHANGES_REQUESTED** [review]
Left some comments. One general point of feedback is that the language could be simplified in several places, for example:

> When leveraging BYOA you will negate Aiven’s feature of paying for all network traffic out of the service.

vs

> Unlike Aiven's standard all-inclusive pricing, with BYOA you will be responsible for all cloud infrastructure and network traffic charges. 

I think the final "Testing, POC, and deployments" section needs work, but because there was no change there in this PR I could not comment in-line. 
- We say to allocate sufficient time but give no indication about how long that should be. hours, days, weeks... months??
- Despite the heading, we can't support BYOA during any kind of POC so maybe we should be clearer about that. 

There's an argument to say this last section shouldn't be in the docs at all. The guidance is too vague to be useful.

File: docs/platform/concepts/byoa.rst [hunk 1]
```diff
@@ -1,50 +1,100 @@
...
+In order to accomplish this, Aiven will utilize a bastion host, logically separated from the
+Aiven services a customer deploys. As the user of these services (e.g. Aiven for Apache Kafka®),
+you will be able to utilize them through standard VPC peering techniques. Although the bastion
+host and the service nodes will reside in a customer managed VPC, they will not be accessible
+(e.g. SSH) to anyone outside of Aiven.
+
+Depending on the service being used, Aiven will take regular backups to enable forking,
+Point inTime Recovery (PITR) and disaster recovery. These backups by default will not
+reside in the customer’s cloud account. If there is a requirement to have all backups
+in your own account we can do this as well. Aiven will need object storage and permissions
```
[the remark this answers is not in the document]
Comment (ak-aiven) [reply docs/platform/concepts/byoa.rst:57]: The only caveat that I know of is that they are still encrypted using our keys. So although the backups will be in their account they won't be able to utilize them. I originally had some comment about BYOK on our roadmap but was told to take it out.

**Comment by ak-aiven** [comment]
I 100% agree about the last section and have completely removed it.

**Pull request merged**